### PR TITLE
Add file_delete method to delete experiment file

### PR DESCRIPTION
**Pull request by 1d4n**
Implemented `file_delete` service function to delete an artifact file of an experiment.

### CODE REVIEW

File: releasenotes/notes/file-delete-4ca0baa574907813.yaml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+
+features:
+  - |
+    Added `file_delete` method to delete experiment artifacts.
```
Comment (coruscating) [inline releasenotes/notes/file-delete-4ca0baa574907813.yaml:5]: This also needs to be in sphinx format.
```suggestion
    Added the :meth:`~IBMExperimentService.file_delete` method to delete experiment artifacts.
```

File: qiskit_ibm_experiment/service/ibm_experiment_service.py [hunk 2]
```diff
@@ -1742,6 +1742,31 @@ def file_download(
         )
         return file_data
 
+    def file_delete(self, experiment_id: str, file_pathname: str):
+        """Deletes a data file from the DB
+
+        Note:
```
Comment (coruscating) [inline qiskit_ibm_experiment/service/ibm_experiment_service.py:1748]: I don't think this note will render correctly, you'll want to use `.. note::` (consult the [sphinx rst primer](https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html)). You can build the docs locally with `tox -edocs` and preview them.

**Review by coruscating — COMMENTED** [review]
Tested in Qiskit Experiments and this works! Just some minor comments.

File: qiskit_ibm_experiment/service/ibm_experiment_service.py (hunk 2 above)
Comment (1d4n) [reply qiskit_ibm_experiment/service/ibm_experiment_service.py:1748]: It works

Comment (coruscating) [reply qiskit_ibm_experiment/service/ibm_experiment_service.py:1748]: How did you test this? I actually don't see `file_delete()` in the built docs from the [latest run](https://github.com/Qiskit-Extensions/qiskit-ibm-experiment/actions/runs/8735063112?pr=101):
image

Comment (1d4n) [reply qiskit_ibm_experiment/service/ibm_experiment_service.py:1748]: The local `qiskit_ibm_experiment` module is not accessible to sphinx, so it uses the existing pip module, which doesn't have the `file_delete` method.
When I add the path of the local `qiskit_ibm_experiment` to `PYTHONPATH `, sphinx builds the documentations for the new methods.

**Pull request merged**